src/com/mambu/apisdk/util/RequestExecutorImpl.java

```diff
@@ -165,6 +141,284 @@ public String executeRequest(String urlString, ParamsMap params, Method method,
 		return response;
 	}
 
+	/*
+	 * Executes a request in order to download content and returns it as a ByteArrayOutputStream
+	 */
+	@Override
+	public ByteArrayOutputStream executeRequest(String urlString, ParamsMap params, ApiDefinition apiDefinition)
+			throws MambuApiException {
```

Comment:
Right... Yes, If it's done in ServiceExecutor already than it's certainly sufficient. May be just the documentation part for all public methods needed some more details: it's seemed hidden in the implementation itself that the ByteArrayOutputStream can be used ONLY with the ZIP_ARCHIVE return type. 
You may want also to consider in ServiceExecutor using switch statement for the return type (instead of IF), especially if in a future other types could be supported for the ByteArrayOutputStream too (or may be combining this logic with an existent such switch just below)